File: registers/temp_register/temp_register.cc
#include "CAProject/registers/temp_register/temp_register.hh"

namespace gem5
{

TempRegister::TempRegister(const TempRegisterParams &params) : 
    SimObject(params),
    temp(params.temp)
{
}

void
TempRegister::setValue(uint16_t val)
{
    temp = val;
}

void
TempRegister::increment()
{
    temp++;
}

void
TempRegister::clear()
{
    temp = 0;
}

uint16_t
TempRegister::getValue()
{
    return temp;
}

} // namespace gem5

File: registers/temp_register/TempRegister.py
from m5.params import *
from m5.SimObject import SimObject

class TempRegister(SimObject):
    type = 'TempRegister'
    cxx_header = "CAProject/registers/temp_register/temp_register.hh"
    cxx_class = "gem5::TempRegister"

    temp = Param.UInt16(0, "Temporary data stored in the register")
